frontend/packages/topology/src/components/page/TopologyView.tsx

```diff
@@ -160,16 +161,26 @@ export const ConnectedTopologyView: React.FC<ComponentProps> = ({
   const searchParams = queryParams.get('searchQuery');
   const fileTypes = supportedFileExtensions.map((ex) => `.${ex}`).toString();
 
-  const onSelect = React.useCallback((entity?: GraphElement) => {
-    // set empty selection when selecting the graph
-    const selEntity = isGraph(entity) ? undefined : entity;
-    setSelectedEntity(selEntity);
-    if (!selEntity) {
-      removeQueryArgument('selectId');
-    } else {
-      setQueryArgument('selectId', selEntity.getId());
-    }
-  }, []);
+  const onSelect = React.useCallback(
+    (entity?: GraphElement) => {
+      if (!entity) {
+        sessionStorage.removeItem(LAST_TOPOLOGY_OVERVIEW_OPEN_STORAGE_KEY);
+      }
+      // set empty selection when selecting the graph
+      const selEntity = isGraph(entity) ? undefined : entity;
+      setSelectedEntity(selEntity);
+      if (!selEntity) {
+        removeQueryArgument('selectId');
+      } else {
```

Comment:
I think your `sessionStorage.removeItem` should be part of the `if (!selEntity)` check or? At least the comment sounds like that to me.
```suggestion
      // set empty selection when selecting the graph
      const selEntity = isGraph(entity) ? undefined : entity;
      setSelectedEntity(selEntity);
      if (!selEntity) {
        sessionStorage.removeItem(LAST_TOPOLOGY_OVERVIEW_OPEN_STORAGE_KEY);
        removeQueryArgument('selectId');
      } else {
```

Reply:
Yes, my bad. I updated it. PTAL. Thanks